RestartManager.js

```diff
@@ -25,6 +26,11 @@ const RestartManager = (() => {
     }
 
     function restartApp(onlyIfUpdateIsPending = false) {
+        if (_in_progress) {
+            log("A restart request is already in progress or queued");
+            return;
+        }
+        _in_progress = true;
```

Comment:
If there isn't a pending update, and an app calls this twice specifying `true`, for the `onlyIfUpdateIsPending` parameter, then this wouldn't be reset to `false` since the app wouldn't actually restart. I don't think that's the end of the world, but since we're logging messages when a restart is "blocked" due to one being in progress, I'd prefer to address this edge case.